doc/openapi/python/cleaner.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from collections import defaultdict
+from dataclasses import dataclass, field
+from ruamel.yaml import YAML
+
+yaml = YAML()
+
+
+@dataclass
+class Schema:
+    uses: set = field(default_factory=set)
+    used_by: set = field(default_factory=set)
+
+
+@dataclass
+class SchemaCleaner:
+    """
+    Implementation of a schema cleaner to remove unused schema definitions.
+
+    Note: This is a one time use object, do not re-use it!
+    """
+    yml: any
+    schemas: dict[str, Schema] = field(default_factory=lambda: defaultdict(Schema))
```

Comment:
to get it to work, i had to add a quote around the type:

    schemas: 'dict[str, Schema]' = field(default_factory=lambda: defaultdict(Schema))
    unused: 'set[str]' = field(default_factory=set)